compiler/rustc_middle/src/ty/context.rs

```diff
@@ -877,6 +877,26 @@ impl<'tcx> TyCtxt<'tcx> {
         self.get_lang_items(())
     }
 
+    pub fn default_traits(self) -> &'static [rustc_hir::LangItem] {
+        match self.sess.opts.unstable_opts.experimental_default_bounds {
```

Comment:
Why is there a -Z flag and a feature gate? Could the feature gate enable both? or is that the part that was mentioned with being too annoying with `cfg(bootstrap)`? (though I don't see how it would actually be an issue here, as the no_core tests can just use the feature gate, too, and libcore would never add it.

Reply:
It is much easier to opt-in default bounds for all extern code using a flag instead of enabling feature for all crates.

Reply:
a little more detailed:

After landing this patch in beta we would be able to define the appropriate lang items in core and opt-in default bound for all rustc crates by adding `-Zexperimental_default_bounds` in `builder.rs` instead of adding `#[feature(...)]` everywhere. This way we can freely add `?Trait` bounds without `cfg(bootstrap)` in core/library and found a scheme for safe and backward compatible with old editions support for `Leak` and `SyncDrop` types.